### PR TITLE
high resolution for I2S input/output, added S/PDIF input/output with configurable sample rate and 24 bit

### PR DESCRIPTION
**Pull request by IDC-Dragon**
Following up on the work of [jcj83429](https://github.com/jcj83429)'s pull request https://github.com/chipaudette/OpenAudio_ArduinoLibrary/pull/8 which got stale:
This transfers I2S input and output with 32 bit, preserving highres audio. 
I have also added "floatified" versions oft the current S/PDIF inputs and output from the "standard" audio library, here with 24 bit precision and configurable sample rate.
In other words, the float library now finally has the hardware inputs and outputs it deserves, to work beyond 44.1kHz and 16 bit.
I have tested all the modified I/Os with a Teensy 4.0, including small signals below 16 bit.
The commits are rather "minimum change", personally I would tend to clean this up and optimize, try to reduce the ISR work and locking time.

### CODE REVIEW

**Comment by boblark** [comment]
This is good stuff.  Thanks much for the contribution.  Now I need to get get some optical cable hardware to use the S/PDIF!  Cheers, Bob
